### ISSUE TITLE
Cannot reproduce the performance as the pretrained model

### ISSUE THREAD

**Issue by mo95** [yangxy/GPEN#55]
Hi Tao,

Thanks for your interesting work with promising results. I have done lots of attempts to reproduce this work with the same settings as that in the paper but I cannot get the performance as robust as the pretrained models. Actually, my reproduced models are much worse than your pretrained models, especially for inputs with heavy degradations (but can get similar results to another GAN prior-based method, i.e., [GFPGAN](https://github.com/TencentARC/GFPGAN)). 

I don't know which parts brought this difference significantly, could you give me some advice on that?  Many thanks!

**Comment by yangxy** [comment]
Actually, the key to reproduce GPEN is to train it like a GAN with an extra L1 loss. You can try it in this way. This answer is also to #24.

**Comment by mo95** [comment]
What does the extra L1 loss mean? Actually, I used the L1 loss and perceptual loss as the reconstruction losses for training. Besides, the weights for different losses follow the settings described in the paper.

**Comment by lisc199** [comment]
@mo95 I tried to reproduce this work. I applied the settings mentioned in the paper. And I found a strange phenomenon is that the   generator's loss drops to a very small value after a few iterations. And it outputs input low quality image with artifacts. Do you meet this problem and how to resolve it?

**Comment by lisc199** [comment]
> @mo95 I tried to reproduce this work. I applied the settings mentioned in the paper. The training collapsed. The generator cannot generator normal faces.
> And if I fixing the decoder and discriminator, I found a strange phenomenon is that the generator's loss drops to a very small value after a few iterations. And it outputs input low quality image with artifacts. Do you meet this problem and how to resolve it?

I added other loss such as vgg loss/region loss(mouth, eyes, etc.). I can get a good result but worse than GPEN.

**Comment by xuewengeophysics** [comment]
> Hi Tao,
> 
> Thanks for your interesting work with promising results. I have done lots of attempts to reproduce this work with the same settings as that in the paper but I cannot get the performance as robust as the pretrained models. Actually, my reproduced models are much worse than your pretrained models, especially for inputs with heavy degradations (but can get similar results to another GAN prior-based method, i.e., [GFPGAN](https://github.com/TencentARC/GFPGAN)).
> 
> I don't know which parts brought this difference significantly, could you give me some advice on that? Many thanks!

I tried GFP-GAN, the results is ok. GPEN has some advantages, I want to apply them to GFP-GAN. Can you leave your email or WeChat to me? Thanks.
